flink-datastream/src/main/java/org/apache/flink/datastream/impl/context/AllKeysContext.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.datastream.impl.context;
+
+import java.util.Iterator;
+
+/**
+ * This class maintains the context of all input keys. It will be notified once a key is selected.
+ */
+public interface AllKeysContext {
```

Comment:
1. Why making this an interface?
2. Wouldn't it be good enough to add a HashSet in ProcessOperator and override setKeyContext?